### PR TITLE
Change the InjectedWeb3Subprovider to accept Web3.Provider

### PR DESCRIPTION
**Pull request by dekz**
## Description

Remove the requirement to pass in a Web3 object to InjectedWeb3Subprovider and accept a Web3 Provider.

## Motivation and Context

Allows us to use a controlled version of Web3.

## How Has This Been Tested?

Manually tested in Website which uses InjectedWeb3 provider.

## Types of changes

* [x] Change requires a change to the documentation.
* [ ] Added tests to cover my changes.

### CODE REVIEW

File: packages/subproviders/src/subproviders/injected_web3.ts [hunk 1]
```diff
@@ -1,17 +1,16 @@
 import * as _ from 'lodash';
 import Web3 = require('web3');
-import Web3ProviderEngine = require('web3-provider-engine');
 
 /*
  * This class implements the web3-provider-engine subprovider interface and forwards
  * requests involving user accounts (getAccounts, sendTransaction, etc...) to the injected
- * web3 instance in their browser.
+ * MetamaskInpageProvider instance in their browser.
```
Comment (fabioberger) [inline packages/subproviders/src/subproviders/injected_web3.ts]: Let's just say the `injected provider instance`, and not mention Metamask specifically. It could be any provider.

File: packages/subproviders/src/subproviders/injected_web3.ts [hunk 2]
```diff
@@ -43,7 +42,7 @@ export class InjectedWeb3Subprovider {
     }
     // Required to implement this method despite not needing it for this subprovider
     // tslint:disable-next-line:prefer-function-over-method
-    public setEngine(engine: Web3ProviderEngine) {
+    public setEngine(engine: any) {
```
Comment (fabioberger) [inline packages/subproviders/src/subproviders/injected_web3.ts]: Why was this changed?

File: packages/website/ts/blockchain.ts [hunk 3]
```diff
@@ -100,7 +100,7 @@ export class Blockchain {
             // We catch all requests involving a users account and send it to the injectedWeb3
             // instance. All other requests go to the public hosted node.
             provider = new ProviderEngine();
-            provider.addProvider(new InjectedWeb3Subprovider(injectedWeb3));
+            provider.addProvider(new InjectedWeb3Subprovider(injectedWeb3.currentProvider));
             provider.addProvider(new FilterSubprovider());
             provider.addProvider(new RedundantRPCSubprovider(publicNodeUrlsIfExistsForNetworkId));
             provider.start();
```
Comment (fabioberger) [inline packages/website/ts/blockchain.ts]: Can you also start a PR on wiki repo updating the wiki articles with this change?

File: packages/subproviders/src/subproviders/injected_web3.ts (hunk 2 above)
Comment (dekz) [reply packages/subproviders/src/subproviders/injected_web3.ts]: Since we're not using the type in any way its only use was to be imported and then referenced here. By removing the type and using any in a method that is a noop, we can remove the import of the Web3 Provider Engine in this class.

File: packages/subproviders/src/subproviders/injected_web3.ts [hunk 4]
```diff
@@ -42,8 +41,9 @@ export class InjectedWeb3Subprovider {
         }
     }
     // Required to implement this method despite not needing it for this subprovider
+    // This type is Web3ProviderEngine, but there is no need to import this for a noop.
```
Comment (fabioberger) [inline packages/subproviders/src/subproviders/injected_web3.ts]: // The `engine` argument type should be `Web3ProviderEngine` but we've decided to keep it as type `any` to // remove the providerEngine dependency given that this method is a noop anyways
